test/tests/bcs/ad_vector_function_neumann_bc/tests

```diff
@@ -0,0 +1,17 @@
+[Tests]
+    issues = '#22327'
+    requirement = 'The system shall support Neumann type boundary conditions defined by a function'
+    [./test]
+        type = 'Exodiff'
+        input = 'vector_ad_neumann_bc.i'
+        exodiff = 'vector_ad_neumann_bc_out.e'
+        detail = 'for vector variables with automatic differentiation and'
+        design = 'ADVectorFunctionNeumannBC.md'
+    [../]
+    [./test-jac]
+        type = 'PetscJacobianTester'
+        input = 'vector_ad_neumann_bc.i'
+        run_sim = False
+        detail = 'be able to produce the exact Jacobian.'
+      [../]
+[]
```

Comment:
```suggestion
[Tests]
  issues = '#22327'
  design = 'ADVectorFunctionNeumannBC.md'
  [vector]
    requirement = 'The system shall support Neumann type boundary conditions defined by a function'
    [test]
      type = 'Exodiff'
      input = 'vector_ad_neumann_bc.i'
      exodiff = 'vector_ad_neumann_bc_out.e'
      detail = 'for vector variables with automatic differentiation and'
    []
    [test-jac]
      type = 'PetscJacobianTester'
      input = 'vector_ad_neumann_bc.i'
      run_sim = False
      detail = 'be able to produce the exact Jacobian.'
    []
  []
[]
```

Alright, the SQA check is being finicky....I had to build this locally to figure this one out and a few sub-block was required. Should be good with this change.

Reply:
Thanks for the help!